app/src/main/java/com/scvsoft/wipeer/NetworkService.java

```
@@ -363,6 +363,7 @@ public void onFailure(int reason) {
     private void connectP2p(WiFiP2pService service) {
 
         if (currentStatus == NetworkStatus.CONNECTING || currentStatus == NetworkStatus.CONNECTED) {
+            Log.d(TAG, "A connection was attempeted while the device was on status " + currentStatus);
             return;
```

I'm not sure this is a valid scenario, the logging is just for debugging purposes if it ever happens. The idea is that if you're connecting or connected a new attempt is silently ignored and logged, not to disrupt the interaction (an exception should be handled with a message to the user but this is not something the user did wrong).